### PR TITLE
[W5.7][W10-1]Kong Zijin

### PR DESCRIPTION
**Pull request by kzzj217**
1. Add a new command "sort" to addressbook-level2.

### CODE REVIEW

**Comment by alistairong** [comment]
Good documentation. Did not need to add method to command class nor person class, as they are not directly affected by the creation of a sortcommand class (creation of a child class should not change parent class much). Test cases for sort command would be good to have as well.

**Comment by xueyantian** [comment]
Good implementation. Could add test cases for both JUnit and IO tests.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.Tag;
+
+/**
+ * Adds a person to the address book.
```
Comment (damithc) [inline src/seedu/addressbook/commands/SortCommand.java:18]: Incorrect comment?

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    /*
+    public SortCommand() throws IllegalValueException {
+       // Done nothing now.
+    }
+    */
```
Comment (damithc) [inline src/seedu/addressbook/commands/SortCommand.java:38]: Don't leave commented out code. Just go ahead and delete unused code. Now that we are using a revision control system, we can recover any deleted code later.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 3]
```diff
@@ -81,6 +81,19 @@ public UniquePersonList(UniquePersonList source) {
         return Collections.unmodifiableList(internalList);
     }
 
+    /**
+     * Returns an anmodifiable java List view with elements cast as immutable
```
Comment (damithc) [inline src/seedu/addressbook/data/person/UniquePersonList.java:85]: Typo: anmodifiable

**Review by damithc — APPROVED** [review]
:+1: Good effort on this PR @KongZijin . In particular, good job on,
* updating docs
* keeping the PR relatively clean


Suggestions to improve further:
* Update I/O tests and JUnit tests

**Comment by damithc** [comment]
As before, please close the PR after viewing comments.

File: src/seedu/addressbook/commands/SortCommand.java (hunk 2 above)
Comment (kzzj217) [reply src/seedu/addressbook/commands/SortCommand.java:38]: Noted with thanks!

**Pull request closed without merging**